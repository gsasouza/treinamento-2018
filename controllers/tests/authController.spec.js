const request = require('supertest');
const app = require('../../app');
const User = require('../../models/userModel');
const { sanitizeTestObject } = require('../../test/sanitizeObject');

beforeAll(() => User.remove({}));

describe('Authenticate', () => {
  test('should authenticate', async () => {
    await User.create({
      username: 'test3',
      password: 'test123',
      name: 'name',
    });

    const response = await request(app).post('/auth').send(
      {
        username: 'test3',
        password: 'test123',
      }
    );
    expect(response.status).toBe(200);
    expect(sanitizeTestObject(response.body, ['token'])).toMatchSnapshot();
  })
});
