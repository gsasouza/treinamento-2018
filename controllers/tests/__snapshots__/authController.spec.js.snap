// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticate should authenticate 1`] = `
Object {
  "message": "Autenticado com sucesso",
  "token": "FROZEN-TOKEN",
}
`;
